agent_build_refactored/scripts/get_github_actions_job_matrices.py

```diff
@@ -149,22 +170,49 @@ def main():
         if is_master_run_only and not master_run:
             continue
 
-        # Set default valued for some essential matrix values, if not specified.
-        if "os" not in job:
-            job["os"] = "ubuntu-20.04"
-        if "python-version" not in job:
-            job["python-version"] = IMAGES_PYTHON_VERSION
+        _apply_job_defaults(job)
 
         builder_name = job["name"]
         builder = ALL_IMAGE_BUILDERS[builder_name]
         job["builder-fqdn"] = builder.get_fully_qualified_name()
         result_images_build_matrix["include"].append(job)
         used_runners.append(builder)
 
-    # Get pre-built steps that are used by this workflow and create matrix for a pre-built steps.
-    pre_built_steps = _get_runners_all_pre_built_steps(
-        runners=used_runners
+    k8s_images_test_matrix_json_file_path = pl.Path(
+        args.k8s_images_test_matrix_json_file
     )
+
+    # Generate a final agent k8s image test job matrix and filter out job for non-master run, if needed.
+    k8s_images_test_matrix = json.loads(
+        k8s_images_test_matrix_json_file_path.read_text()
+    )
+    result_k8s_images_test_matrix = {"include": []}
+    for job in k8s_images_test_matrix:
+        is_master_run_only = job.get("master_run_only", True)
+        # If this is non-master run, skip jobs which are not supposed to be in it.
+        if is_master_run_only and not master_run:
+            continue
```

Comment:
Do we want to log a message here which says that job is X skipped because it's set to only run on master and this is not a master running or something along those lines?

I know this script also returns result so we probably need to log to stderr to not pollute stdout.